### PR TITLE
Crash prevention from assert( CFSocketGetSocketFlags(self.socket) & kCFSocketCloseOnInvalidate );

### PR DESCRIPTION
**Pull request by nneuberger1**
We swapped out the assert because we're getting crashes on the following line:

assert( CFSocketGetSocketFlags(self.socket) & kCFSocketCloseOnInvalidate );

Instead we're using if / else if statements and if found present, then we return an error and let the caller get an error back from a ping response on the Swift side.

### CODE REVIEW

**Comment by nneuberger1** [comment]
We've been running my master branch for almost a year now.  Any chance we can merge this into your repo?  We're very stable with change I have on master.

**Comment by seeppp** [comment]
Thanks for testing your changes for almost a year 😉 

I'll merge it! But keep in mind, we don't do any maintenance on this repository anymore. (I probably have to update the Readme...)

**Comment by nneuberger1** [comment]
No problem @seeppp.  We've been using my fork the entire time. We use it for credit card VeriFone e355 swipers. at some point, we're going to migrate to other vendors that have built in rest interfaces.

File: Pod/Classes/SimplePing.m [hunk 1]
```diff
@@ -627,13 +627,23 @@ - (void)startWithHostAddress {
         
         // The socket will now take care of cleaning up our file descriptor.
         
-        assert( CFSocketGetSocketFlags(self.socket) & kCFSocketCloseOnInvalidate );
+        CFSocketSetSocketFlags(self.socket, kCFSocketCloseOnInvalidate);
```
Comment (dstd) [inline Pod/Classes/SimplePing.m:630]: Here for no reason you're erasing all default flags, specifically kCFSocketAutomaticallyReenableReadCallBack, which leads to issues with simultanious instances of SimplePing (#20)

Comment (dstd) [reply Pod/Classes/SimplePing.m:630]:
```suggestion
        CFSocketSetSocketFlags(self.socket, (CFSocketGetSocketFlags(self.socket) | kCFSocketCloseOnInvalidate));
```

**Pull request merged**